autoria.py

```diff
@@ -0,0 +1,253 @@
+"""Python auto.ria.com API.
+
+Python implementation of API intended for calulating
+average used car prices that are sold on http://auto.ria.com
+"""
+
+
+import requests
+import json
+from fnmatch import fnmatch
+from typing import Union
+
+
+class RiaAPI:
+    """Python auto.ria.com API.
+
+    Python implementation of API intended for calulating
+    average used car prices that are sold on http://auto.ria.com
+    """
+
+    def __init__(self):
```

Comment:
> Not sure should I initialize an instance with search parameters, or should I pass them into .get_average() as in was before. Please advice.
> Also I have no idea what should be described in class docstring (just copy-pasted there module docstring), and the same for class __init()__ method.

This is actually a very good question and the answer depends on what do we actually want to do here.

If we just want to have an "average car price" calculation then we put the search parameters into the constructor, so one object will represent one search:

```
my_car_average = RiaAveragePrice(parameter1, parameter2)
your_car_average = RiaAveragePrice(parameterX, paramterY)
```

Note, that the class is named appropriately (`RiaAveragePrice`).
If we want to have an API wrapper that makes all the ria API methods available to the code using the class, then we should keep it as is. For example, in some other case, we might just use a list of bodystyles and nothing else.

Actually I think we are now trying to make this class do both and it hints that it should be split into two classes:

```
   class RiaAPI:
       "Same as now, except the average method - it shouldn't perform any selects, it will just make an API request"

       ...

       def average_price(main_category, body_id, marka_id, ...): 
            return self._make_request('/average', parameters)

```

And the class to provide a "human" interface to the average method:

```
class RiaAverageCarPrice:

    def __init__(self, category: str, mark: str, model: str,
                    bodystyle: str = None, years: list = None,
                    state: str = None) -> None:
        self._api = RiaAPI()
        self._params = {}
        # Getting list of categories and selecting needed id
        categories_list = self._api.get_categories()
        self._params['main_category'] = self.select_item(category, categories_list)

        # Getting list of bodysyles and selecting needed id
        bodystyles_list = self._api.get_bodystyles(selected_category)
        self._params['body_id'] = self._select_item(bodystyle, bodystyles_list)

        ....

    def _select_item(self, ...):
       """Probably select_item method should be in this class and it should be private.
           If it is generally useful, it can be a free function in this module."""

    def get_average() -> Union[dict, list]:
        """Compose parameters for GET request and send it.
        Get average price using search parameters.
        Args:
            ...
        Returns:
                ...
        """
        return self._api.average_price(self._params)
```